aiphysim/dataloading/dataloader_spacetime.py

```diff
@@ -72,6 +66,8 @@ def __init__(
                 ],
                 axis=0,
             )
+            self.data = self.data.astype(np.float32)
```

Comment:
What's the original data type here, and why do we need to convert it to float32?

Reply:
The original type was float64 which is also known as "long". But the model requires a float32 as input, hence the type casting